### PR TITLE
Airdrop page feature

### PR DESCRIPTION
**Pull request by jb1011**
aidrop page added on url (/#/beta/airdrop)
Wagmi airdrop will be available when airdrop allower script is run

### CODE REVIEW

File: src/pages/Beta/Airdrop2/QuestionBox/index.tsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  }
+
+  const [title, setTitle] = useState<string>('')
+  const [content, setContent] = useState<string>('')
+
+  axios(config).then(function(response: any) {
+    setTitle(JSON.stringify(response.data.data.kb.map((e: any) => e.title)).replace(/[[\]"]/g, ''))
+    setContent(JSON.stringify(response.data.data.kb.map((e: any) => e.content)).replace(/[[\]"]/g, ''))
+  })
+
```
Comment (SarjuHansaliya) [inline src/pages/Beta/Airdrop2/QuestionBox/index.tsx:38]: this is not a good practice to call axios on every render. please separate this logic into some hook, please refer how we have implemented news apis

File: src/pages/Beta/Airdrop2/ButtonToConnect/index.tsx [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+import { Button } from '@pangolindex/components'
+import { useWalletModalToggle } from 'src/state/application/hooks'
+import { useTranslation } from 'react-i18next'
+
+export const ButtonToConnect = () => {
+  const toggleWalletModal = useWalletModalToggle()
+  const { t } = useTranslation()
+  return (
+    <Button variant="primary" color="white" height="46px" onClick={toggleWalletModal}>
+      <span style={{ whiteSpace: 'nowrap', color: '#000', fontSize: '20px' }}>{t('swapPage.connectWallet')}</span>
```
Comment (SarjuHansaliya) [inline src/pages/Beta/Airdrop2/ButtonToConnect/index.tsx:11]: lets not use inline style

File: src/pages/Beta/Airdrop2/index.tsx [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+          <Text fontSize={18} fontWeight={500} lineHeight="27px" color="text10">
+            And we are not empty handed!
+          </Text>
+        </span>
+      </Box>
+      <BoxWrapper>
+        {renderBoxesWagmi()}
+        <Confetti start={Boolean(eligible)} />
+        <ClaimBox>
+          <span
```
Comment (SarjuHansaliya) [inline src/pages/Beta/Airdrop2/index.tsx:111]: move all inline styles from this file to separate styled component, please refer how we have implemented other pages. apply this type of change to every file in this PR

File: src/pages/Beta/Airdrop2/styleds.tsx [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+  display: flex;
+  justify-content: center;
+  gap: 15px;
+  margin-bottom: 15px;
+
+  ${({ theme }) => theme.mediaWidth.upToMedium`
+  flex-direction: column;
+  align-items: center;
+  justify-content: center;
+  `};
```
Comment (SarjuHansaliya) [inline src/pages/Beta/Airdrop2/styleds.tsx:28]: please check formatting again in this file

File: src/pages/Beta/Airdrop2/wagmiBoxes/BoxBuyCurrency/index.tsx [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+      setButton1('outline')
+      setButton2('primary')
+      setButton3('outline')
+    } else if (amount === 1) {
+      setButton1('outline')
+      setButton2('outline')
+      setButton3('primary')
+    }
+  }
+  console.log(selected)
```
Comment (SarjuHansaliya) [inline src/pages/Beta/Airdrop2/wagmiBoxes/BoxBuyCurrency/index.tsx:32]: remove console

File: src/pages/Beta/Airdrop2/wagmiBoxes/BoxBuyCurrency/index.tsx [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+        <Button
+          variant={button2}
+          color="white"
+          height="39px"
+          borderColor="white"
+          onClick={() => {
+            selectAmount(0.5)
+          }}
+        >
+          <span style={{ whiteSpace: 'nowrap', color: '#000', fontSize: '15px' }}>0.5 AVAX</span>
```
Comment (SarjuHansaliya) [inline src/pages/Beta/Airdrop2/wagmiBoxes/BoxBuyCurrency/index.tsx:67]: I see that everywhere Button is having span child. is there any specific need for that ? color can be handled using `color` prop and I dont think we generally need to change font size. if not required please remove span from every Button and just directly use Button component

File: src/pages/Beta/Airdrop2/wagmiBoxes/BoxCheckEligibility/index.tsx [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+      </Text>
+      <span style={{ padding: '20px' }}></span>
+      <Button variant="primary" color="white" height="46px" onClick={checkStatus}>
+        <span style={{ whiteSpace: 'nowrap', color: '#000', fontSize: '20px' }}>CHECK IF ELIGIBLE</span>
+      </Button>
+      <span style={{ textAlign: 'center' }}>
+        <Text fontSize={14} fontWeight={500} lineHeight="18px" color="text8">
+          To be eligible or not to be eligible...
+        </Text>
+      </span>
```
Comment (SarjuHansaliya) [inline src/pages/Beta/Airdrop2/wagmiBoxes/BoxCheckEligibility/index.tsx:32]: what is reason to wrap Text with span. I think text should be able to handle textAlign center

**Review by SarjuHansaliya — CHANGES_REQUESTED** [review]

File: src/state/airdrop/hooks.ts [hunk 8]
```diff
@@ -75,3 +77,39 @@ export function useClaimCallback(
...
+  }`,
+    variables: { filter: { category: { _eq: 'Airdrop' } } }
+  })
+
+  const headers = {
+    'Content-Type': 'application/json'
+  }
+
+  return useQuery('getQuestions', async () => {
+    const response = await axios.post('https://p7gm7mqi.directus.app/graphql', queryData, { headers: headers })
```
Comment (SarjuHansaliya) [inline src/state/airdrop/hooks.ts:104]: please move this url to `src/constants/index.ts` on top

**Review by SarjuHansaliya — CHANGES_REQUESTED** [review]
there are few linting errors. please run `npm run lint`
Screenshot 2022-05-09 at 10 20 32 AM

File: src/state/airdrop/hooks.ts (hunk 8 above)
Comment (jb1011) [reply src/state/airdrop/hooks.ts]: URL moved to constant and lint errors have been solved

**Pull request merged**